scipy/stats/_hypotests.py

```diff
@@ -663,3 +663,272 @@ def somersd(x, y=None):
+      with `alternative` = "less"
+
+    - :math:`H_0 : p_1 \leq p_2` versus :math:`H_1 : p_1 > p_2`,
+      with `alternative` = "greater"
+
+    - :math:`H_0 : p_1 = p_2` versus :math:`H_1 : p_1 \neq p_2`,
+      with `alternative` = "two-sided" (default one)
+
+    References
+    ----------
+    .. [1] Mehta, Cyrus & Senchaudhuri, Pralay. (2003).
+           "Conditional versus Unconditional Exact Tests for Comparing Two
+           Binomials"
+
+    Examples
+    --------
+    An example use of Barnard's test is presented in [1]_.
+
+        Consider the following example of a vaccine efficacy study
+        (Chan, 1998). In a randomized clinical trial of 30 subjects, 15 were
+        inoculated with a recombinant DNA influenza vaccine and the 15 were
+        inoculated with a placebo. Twelve of the 15 subjects in the placebo
+        group (80%) eventually became infected with influenza whereas for the
+        vaccine group, only 7 of the 15 subjects (47%) became infected. The
+        data are tabulated as a 2 x 2 table::
+
+                Vaccine  Placebo
+            Yes     7        12
+            No      8        3
+                    15        15
+    When working with statistical hypothesis testing, we usually use a
+    threshold probability or a significance level upon which we decide
+    to reject or not the null hypothesis :math:`H_0`. A commonly used
+    significance level is 5%. In this example, we are using the
+    `alternative` parameter with the "less" option, because the vaccine has
+    either no effect (:math:`H_0` hypothesis) or a positive effect
+    (:math:`H_1` hypothesis). Therefore, under the null hypothesis
+    (:math:`H_0`), the probability :math:`p_2` of catching the virus
+    **without** any vaccine, is either equal or lower than :math:`p_1`,
+    the probability of catching the virus **with** the vaccine.
+    To compute the p-value with Banard test, let's call `barnard_exact` as
+    follow :
+
+    >>> import scipy.stats as stats
+    >>> res = stats.barnard_exact([[7, 12], [8, 3]], alternative="less")
+    >>> res.statistic
+    1.894...
+    >>> res.pvalue
+    0.03407...
+
+    Then, the probability of obtaining test results at least as extreme as the
+    data observed above is around 3.4%. Since our p-value is under our
+    significance level defined above, we reject :math:`H_0`. The vaccine has
+    a positive effect and we have :math:`p_1 \leq p_2`; the probability of
+    contracting the virus if we have been vaccinated is lower than if
+    we are not.
+
+    We can compare with Fisher's exact test which produces an exact
+    p-value of 6.4% :
+
+    >>> _, pvalue = stats.fisher_exact([[7, 12], [8, 3]], alternative="less")
+    >>> pvalue
+    0.0640...
+
+    With the same threshold probability of 5%, `fisher_exact` is accepting
+    :math:`H_0` while `barnard_exact` is rejecting it. As stated in [1]_,
+    Barnard's test is uniformly more powerful than Fisher exact test.
+    """
```

Comment:
From this, I am not sure I understand why this is a better test. I suggest to add the following:

```suggestion
    With the same threshold probability of 5%, `fisher_exact` is accepting
    :math:`H_0` while `barnard_exact` is rejecting it. As stated in [1]_,
    Barnard's test is uniformly more powerful than Fisher's exact test
    because Barnard's test do not condition on any margin. Fisher's test
    should only be used when both sets of marginals are fixed.

    """
```

Reply:
Thank-you for this add :-)